packages/docs/src/pg/src/plasma/design.rst

```diff
@@ -0,0 +1,166 @@
+
+A downside is *Large token denominations* 
+
+Because each token must be assigned a serial number, one cannot mint arbitrarily small tokens. 
+This is because at some point the gas cost of redeeming the token will be larger than the value of the token itself.
+
+For every block in the Plasma Cash chain, a merkle root must be published to the root chain. This root can either be a 
+merklized list, or a merkle patricia tree. In the merklized list, each index of the leaf nodes corresponds to the token ID. 
+The values of the leaf nodes are Plasma transactions.
+
+Transactions take the form of:
+
+``[[prev_hash, prev_block, (target_block?), token_id, new_owner], signature]``
+
+A transaction spending a token with a given token_id is only valid if it is included in the Merkle tree at position 
+token_id; that is, for each token, there is only one "place" in the Merkle tree where transactions spending that token are 
+allowed to be.
+
+**Deposits**
+
+Anyone can call a deposit() function of the Plasma contract, which mints a new token into existence with the amount of 
+ether sent along with the call as the denomination;
+
+**Withdrawals**
+
+*Starting an Exit*
+
+When a user wants to withdraw a token, they need to submit the two latest transactions in the token’s history. For example, 
+if C wants to withdraw token #4, they need to provide the “child” (most recent) transaction from B to C, and the “parent” 
+transaction from A to B. The user also needs to submit Merkle proofs that show both transactions were included in the 
+blockchain.
+
+*Challenging exits*
+We need to support three types of challenges to ensure that only the true owner of a token can withdraw that token. 
+Withdrawals can be immediately blocked if someone proves that the withdrawing user actually spent the token later on. 
+Withdrawals can also be immediately blocked if someone shows that there’s a transaction between the parent and the child 
+transactions, meaning the withdrawing user provided an invalid parent.
+
+Someone can also challenge the withdrawal by providing some other transaction in the token’s history. This type of 
+challenge doesn’t immediately block a withdrawal. Instead, the withdrawing user is forced to respond with the transaction 
+that comes after the provided transaction.
+
+*Exits*
+Anyone can exit their coin by providing the last two transactions in the coin’s ownership history (ie. the coin they are 
+exiting C and its parent P( C )).
+
+An exit can be challenged in three ways: (i) provide a proof of a transaction spending C, (ii) provide a proof of a 
+transaction spending P( C ) that appears before C, (iii) provide a transaction C* in the coin’s history before P( C )
+
+A challenge of type (i) and (ii) blocks the exit immediately. A challenge of type (iii) can be responded to by providing 
+the direct child of C*, which must be either equal to or before P( C )
+
+Would coin ID => denomination be stored as a mapping on the root chain? 
```

Comment:
@nanspro These lines here I don't entirely understand, 151 down to 158.

Reply:
yeah they were kind of misplaced and the non inclusion stuff is related to SMT's which has already been explained above so i removed that part.